### PR TITLE
[JENKINS-36384] Fix stage viewport scrolling all the way to the left on re-render with many stages

### PR DESCRIPTION
**Pull request by svanoort**
Solves [JENKINS-36384](https://issues.jenkins-ci.org/browse/JENKINS-36384).

@reviewbybees

### CODE REVIEW

**Comment by jglick** [comment]
uncomprehending :bee:

File: ui/src/main/js/view/pipeline-staged.js [hunk 1]
```diff
@@ -54,15 +54,25 @@ function _render(jobRunsData, onElement, fragCaption) {
...
         var runGroup = jobRunsData.runGroups[0];
 
         runGroup.fragCaption = fragCaption;
         runGroup.maxTableEms = runGroup.stageData.length * 10;
 
         var pipelineStagedDom = templates.apply('pipeline-staged', runGroup);
-
         addLaneCharts(pipelineStagedDom, runGroup);
-        onElement.empty().append(pipelineStagedDom);
+        var viewPort = $('div.table-viewPort');
```
Comment (recena) [inline ui/src/main/js/view/pipeline-staged.js]: Why `var viewPort = $('div.table-viewPort:first');`?

File: ui/src/main/js/view/pipeline-staged.js [hunk 2]
```diff
@@ -54,15 +54,25 @@ function _render(jobRunsData, onElement, fragCaption) {
...
+        if (viewPort && viewPort.size() > 0) { // First rendering
+            var leftScroll = viewPort[0].scrollLeft;
+            onElement.empty().append(pipelineStagedDom);
+            viewPort = $('div.table-viewPort')[0];
```
Comment (recena) [inline ui/src/main/js/view/pipeline-staged.js]: Do you need to get it again?

**Comment by rsandell** [comment]
:bee: :baby:

File: ui/src/main/js/view/pipeline-staged.js (hunk 1 above)
Comment (svanoort) [inline ui/src/main/js/view/pipeline-staged.js]: Does that return the single element, or an array that always has a single element (as with the existing one)?  I'm not sure how smart the JQuery selectors are here.

File: ui/src/main/js/view/pipeline-staged.js (hunk 2 above)
Comment (svanoort) [inline ui/src/main/js/view/pipeline-staged.js]: Yes, the DOM has mutated and the original element no longer applies (latest commit adds an explanatory comment).

**Comment by recena** [comment]
🐝

**Pull request merged**